django/thunderstore/social/api/experimental/views/current_user.py

```diff
@@ -96,5 +98,8 @@ def get_subscription_status(user: Optional[UserType]) -> SubscriptionStatus:
     if not user:
         return {"expires": None}
 
-    # TODO: return expiration date if user has active subscription.
+    now = datetime.datetime.now()
+    if "cyberstorm_beta_access" in UserFlag.get_active_flags_on_user(user, now):
+        return {"expires": (now + datetime.timedelta(weeks=4))}
```

Comment:
You could also query the expiry time from the `UserFlag` object directly, but that's a bit more complicated and ultimately doesn't matter much here. The `get_active_flags_on_user` API is intended for checks that need the status of _right now_, whereas the expiry time is about future status.

Anyway feel free to keep this as is IMO, just figured I'd note on this anyway

Reply:
In the beginning I implemented in that way, but decided to uncomplicate it. For this iteration we kinda just need the sub on/off state, because the nextjs middleware only makes a `min(15minutes, sourceTimeStamp)` token. When we have the "not this version" of sub management, then we can make it send the actual end time.